PerimeterXModule/PxModule.cs

```diff
@@ -366,6 +383,33 @@ public void Dispose()
 			}
 		}
 
+
+		/// <summary>
+		/// Checks the url if it should be a proxy request for the client/xhrs
+		/// </summary>
+		/// <param name="context">HTTP context for client</param>
+		private bool IsProxyRequest(HttpContext context)
+		{
+			PxModuleConfigurationSection config = (PxModuleConfigurationSection)ConfigurationManager.GetSection(PxConstants.CONFIG_SECTION);
+			if (ReverseProxy.ShouldReverseClient(context.Request.Url.AbsolutePath))
+			{
+				Debug.WriteLine("Matched client reverse prefix");
+				ReverseProxy.ReversePxClient(context);
+				context.ApplicationInstance.CompleteRequest();
```

Comment:
If both `ReverseProxy.ReversePx*` functions get the `context`, why not do the `context.ApplicationInstance.CompleteRequest()` inside? You can move the debug printouts inside as well. 

Moreover, since all 4 methods reside in `ReverseProxy`, then `ReverseProxy.ReversePxClient` can be responsible for checking `ReverseProxy.ShouldReverseClient` internally, and similarly, `ReverseProxy.ReversePxXhr` can be responsible for `ReverseProxy.ShouldReverseXhr` inside. 

If you follow along, you will get something like -

```
		private bool IsProxyRequest(HttpContext context)
		{
			PxModuleConfigurationSection config = (PxModuleConfigurationSection)ConfigurationManager.GetSection(PxConstants.CONFIG_SECTION);
			bool reversedClient = ReverseProxy.ReversePxClient(context);
			return reversedClient || ReverseProxy.ReversePxXhr(context);
		}
```

Reply:
👍